This patch validates codec and context before trying to call ffmpeg to avoid
crashes when mad is not enabled and an mp3 is tried to be played. This way, if
mad support is not enabled it can fail "gracefully" without having a segfault
that takes down the frontend entirely.

Thanks to Ian Monroe <ian@monroe.nu> for the backtrace from which I started
testing.
Index: xine-lib-1.1.2cvs20060328/src/libffmpeg/audio_decoder.c
===================================================================
--- xine-lib-1.1.2cvs20060328.orig/src/libffmpeg/audio_decoder.c
+++ xine-lib-1.1.2cvs20060328/src/libffmpeg/audio_decoder.c
@@ -242,6 +242,13 @@ static void ff_audio_decode_data (audio_
   } else if (!(buf->decoder_flags & BUF_FLAG_SPECIAL)) {
 
     if( !this->decoder_ok ) {
+      if ( ! this->context || ! this->codec ) {
+        xprintf (this->stream->xine, XINE_VERBOSITY_LOG,
+		_("ffmpeg_audio_dec: trying to open null codec\n"));
+	_x_stream_info_set(this->stream, XINE_STREAM_INFO_AUDIO_HANDLED, 0);
+	return;
+      }
+
       pthread_mutex_lock (&ffmpeg_lock);
       if (avcodec_open (this->context, this->codec) < 0) {
 	pthread_mutex_unlock (&ffmpeg_lock);
